#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- bash/doc/bash.1~	2006-07-12 20:16:36.000000000 -0400
+++ bash/doc/bash.1	2006-07-12 20:26:49.000000000 -0400
@@ -3014,6 +3014,10 @@
 Redirections using file descriptors greater than 9 should be used with
 care, as they may conflict with file descriptors the shell uses
 internally.
+.PP
+Note that the
+.B exec
+builtin command can make redirections take effect in the current shell.
 .SS Redirecting Input
 .PP
 Redirection of input causes the file whose name results from

